parser/prism/Translator.cc

```diff
@@ -1016,21 +1016,36 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto requireds = absl::MakeSpan(paramsNode->requireds.nodes, paramsNode->requireds.size);
             auto optionals = absl::MakeSpan(paramsNode->optionals.nodes, paramsNode->optionals.size);
             auto keywords = absl::MakeSpan(paramsNode->keywords.nodes, paramsNode->keywords.size);
+            auto posts = absl::MakeSpan(paramsNode->posts.nodes, paramsNode->posts.size);
 
             parser::NodeVec params;
 
             auto restSize = paramsNode->rest == nullptr ? 0 : 1;
             auto kwrestSize = paramsNode->keyword_rest == nullptr ? 0 : 1;
             auto blockSize = paramsNode->block == nullptr ? 0 : 1;
 
-            params.reserve(requireds.size() + optionals.size() + restSize + keywords.size() + kwrestSize + blockSize);
+            params.reserve(requireds.size() + optionals.size() + restSize + posts.size() + keywords.size() +
+                           kwrestSize + blockSize);
 
             translateMultiInto(params, requireds);
             translateMultiInto(params, optionals);
 
             if (paramsNode->rest != nullptr)
                 params.emplace_back(translate(paramsNode->rest));
 
+            // Note: This is actually a divergence from the legacy Sorbet parser behavior, which doesn't handle `**nil`
+            //   as a post argument. In this case, the legacy parser would just replace the entire tree with `Nil`.
+            for (auto &post : posts) {
+                switch (PM_NODE_TYPE(post)) {
+                    case PM_NO_KEYWORDS_PARAMETER_NODE: { // `**nil`
+                        params.emplace_back(translateSimpleKeyword<parser::Kwnilarg>(post));
+                        break;
+                    }
+                    default:
+                        params.emplace_back(translate(post));
+                }
+            }
```

Comment:
@st0012 I'm curious to hear your thoughts on this --

In the case where `**nil` is passed as a post argument (e.g. `def foo(**nil, **nil); end`), the legacy parser and Prism have different behavior:

- The Sorbet parser will [replace the entire tree with a `Nil` node](https://sorbet.run/?arg=--print&arg=parse-tree#%23%20typed%3A%20true%0A%0Adef%20foo%28**nil%2C%20**nil%29%3B%20end%0A)
- Prism will replace the `**nil` post parameter with a `NoKeywordsParameterNode` but leave the rest of the tree the same.

I added handling here to prevent some of the tests from crashing, but it introduces a difference in the parse trees.

Are you okay leaving this as is for now and revisiting once we do more error handling work?